src/cs-core/GuiManager.cpp

```diff
@@ -81,13 +80,13 @@ GuiManager::GuiManager(std::shared_ptr<Settings> settings,
         static_cast<float>(mSettings->mGuiPosition->mPosZMeter));
 
     // Create the global GUI area.
-    mGlobalGuiArea = std::make_unique<gui::WorldSpaceGuiArea>(
+    mGlobalGuiArea = std::make_shared<gui::WorldSpaceGuiArea>(
         mSettings->mGuiPosition->mWidthPixel, mSettings->mGuiPosition->mHeightPixel);
     mGlobalGuiArea->setUseLinearDepthBuffer(true);
   }
 
   // Create the local GUI area.
-  mLocalGuiArea = std::make_unique<gui::ScreenSpaceGuiArea>(pViewport);
+  mLocalGuiArea = std::make_shared<gui::ScreenSpaceGuiArea>(pViewport);
```

Comment:
What is the reasoning behind changing from `unique_ptr` to `shared_ptr`? The[ C++ Core Guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-unique) recommend using `unique_ptr` over `shared_ptr`.

Reply:
The these members can now be accessed through the interface of the `GuiManager`. Therefore I changed this.

As the `GuiManager` actually should own these, this is semantically not ideal. What should be done in this case? [Stackoverflow](https://stackoverflow.com/questions/43400850/return-a-unique-ptr-by-reference) suggests using raw pointers. Returning a reference to the `unique_ptr` seems weird, maybe we could simply return a `gui::ScreenSpaceGuiArea const&`?

Reply:
We can use references.

Otherwise we can use an `observer_ptr` (https://github.com/martinmoene/observer-ptr-lite), which is basically a raw pointer disguised as a smart pointer. As far as I know it might be standardized some time in the future.
https://en.cppreference.com/w/cpp/experimental/observer_ptr